autoload/go/test.vim

```diff
@@ -31,7 +34,8 @@ function! go#test#Test(bang, compile, ...) abort
   endif
 
   if go#config#TermEnabled()
-    call go#term#new(a:bang, ["go"] + args, s:errorformat())
+    let l:cmd = "go " . join(args)
+    call go#term#new(a:bang, l:cmd, s:errorformat())
```

Comment:
A list is the right thing to pass. This should also be reverted.

Reply:
@bhcleek 

go#cmd#RunTerm passes a string
```
  let cmd = "go run "
  let tags = go#config#BuildTags()
  if len(tags) > 0
    let cmd .= "-tags " . go#util#Shellescape(tags) . " "
  endif

  if empty(a:files)
    let cmd .= go#util#Shelljoin(go#tool#Files())
  else
    let cmd .= go#util#Shelljoin(map(copy(a:files), "expand(v:val)"), 1)
  endif
  call go#term#newmode(a:bang, cmd, s:runerrorformat(), a:mode)
```

Is the intention to have GoRun pass a string and GoTest pass an array to #newmode ?

Reply:
I believe that picking one argument type is the correct thing and that the one argument type should be a list, not a string. I realize that `go#cmd#RunTerm` passes a string - let's make it a list (`go#test#Test` already uses the list form).

Reply:
sounds good. Will do this.